frame/support/procedural/src/construct_runtime/expand/metadata.rs

```diff
@@ -86,15 +88,32 @@ pub fn expand_runtime_metadata(
 								>::SignedExtensions as #scrate::sp_runtime::traits::SignedExtension
 							>::metadata()
 								.into_iter()
-								.map(|meta| #scrate::metadata::SignedExtensionMetadata {
+								.map(|meta| #scrate::metadata_ir::SignedExtensionMetadataIR {
 									identifier: meta.identifier,
 									ty: meta.ty,
 									additional_signed: meta.additional_signed,
 								})
 								.collect(),
 					},
-					#scrate::scale_info::meta_type::<#runtime>()
-				).into()
+					ty: #scrate::scale_info::meta_type::<#runtime>()
+				}
+			}
+
+			pub fn metadata() -> #scrate::metadata::RuntimeMetadataPrefixed {
+				// Note: `metadata-v14` feature active by default until we stabilize v15,
+				// to keep backwards compatibility.
+				let v14: #scrate::metadata::RuntimeMetadataV14 = #runtime::metadata_ir().into();
+				v14.into()
```

Comment:
The feature is not active by default and we should not assume this! 

Please also make this some function call in `metadata_ir`.

Reply:
I've moved this to the metadata_ir, thanks!